### PR TITLE
Tower shooter icons fix

### LINKED ISSUE #791: wrong icons of arrow towers
Conflux towers have storm els, but in the turn order they have magma els images (the 3x1 stacks): https://imgur.com/a/QgvV8Gb

haven't checked other towns.

### PR DESCRIPTION
**Pull request by IvanSavenko**
Fixes #791 as described in comments.

- added new icons for tower shooters, to be used as fallback in scenario when user plays without vcmi-extra mod
- added functionality to define tower icon per town, allowing icons that represent this town

### CODE REVIEW

File: client/battle/CBattleInterfaceClasses.cpp [hunk 1]
```diff
@@ -763,7 +763,13 @@ void CStackQueue::update()
 		stackBoxes[boxIndex]->setUnit(nullptr);
 }
 
-CStackQueue::StackBox::StackBox(CStackQueue * owner)
+int32_t CStackQueue::getSiegeShooterIconID()
+{
+		return owner->siegeH->town->town->faction->index;
```
Comment (Nordsoft91) [inline client/battle/CBattleInterfaceClasses.cpp:768]:
```suggestion
	return owner->siegeH->town->town->faction->index;
```

File: mapeditor/graphics.cpp [hunk 2]
```diff
@@ -310,11 +310,13 @@ std::shared_ptr<Animation> Graphics::getAnimation(const std::shared_ptr<const Ob
 	return ret;
 }
 
-void Graphics::addImageListEntry(size_t index, const std::string & listName, const std::string & imageName)
+void Graphics::addImageListEntry(size_t index, size_t group, const std::string & listName, const std::string & imageName)
 {
 	if (!imageName.empty())
 	{
 		JsonNode entry;
+		if ( group != 0)
```
Comment (Nordsoft91) [inline mapeditor/graphics.cpp:318]:
```suggestion
		if(group != 0)
```

**Review by Nordsoft91 — COMMENTED** [review]
Where did you get those new assets? Shouldn't they be contained in vcmi-extras instead?

**Comment by IvanSavenko** [comment]
> Where did you get those new assets? Shouldn't they be contained in vcmi-extras instead?
Made by me in gimp. 
Background for larger image is based on public-domain texture that we also have in vcmi-assets repository.

**Comment by Nordsoft91** [comment]
Please bump version of vcmi essential mod

**Review by Nordsoft91 — APPROVED** [review]

**Comment by nullkiller** [comment]
Do we already have new icons in extras mod for regular towns? How can I see them?

**Comment by IvanSavenko** [comment]
There is "Arrow Tower Icons" submod in vcmi extra mod.
After this pull request the submod needs to be updated in order to take use of newly added possibility to define icons per-town.
Without an update the submod will be broken. It won't to any errors/crashes but it won't provide any functionality either.

I assume that either @misiokles or somebody from modders will have to make alternative, per-town icons and once those are done - submod will be updated.

**Comment by misiokles** [comment]
Ok, I'll start to look into and prepare first version os such submod in next week.

**Pull request merged**